threescale/types.go

```diff
@@ -163,43 +140,14 @@ type Transaction struct {
 	Timestamp string
 }
 
-// ***** XML return types from 3scale API
-
-// ApiAuthResponseXML formatted response from backend API for Authorize and AuthRep
-type ApiAuthResponseXML struct {
-	Name         xml.Name     `xml:",any"`
-	Authorized   bool         `xml:"authorized,omitempty"`
-	Reason       string       `xml:"reason,omitempty"`
-	Code         string       `xml:"code,attr,omitempty"`
-	Hierarchy    HierarchyXML `xml:"hierarchy"`
-	UsageReports struct {
-		Reports []UsageReportXML `xml:"usage_report"`
-	} `xml:"usage_reports"`
-}
-
-// HierarchyXML encapsulates the return value when using "hierarchy" extension
-type HierarchyXML struct {
-	Metric []struct {
-		Name     string `xml:"name,attr"`
-		Children string `xml:"children,attr"`
-	} `xml:"metric"`
-}
-
-// UsageReportXML captures the XML response for rate limiting details
-type UsageReportXML struct {
-	Metric       string      `xml:"metric,attr"`
-	Period       LimitPeriod `xml:"period,attr"`
-	PeriodStart  string      `xml:"period_start"`
-	PeriodEnd    string      `xml:"period_end"`
-	MaxValue     int         `xml:"max_value"`
-	CurrentValue int         `xml:"current_value"`
-}
-
-// ReportErrorXML captures the XML response from Report endpoint when not status 202
-type ReportErrorXML struct {
-	XMLName xml.Name `xml:"error"`
-	Text    string   `xml:",chardata"`
-	Code    string   `xml:"code,attr"`
+// UsageReport for rate limiting information gathered from using extensions
+type UsageReport struct {
```

Comment:
Isn't this the same one as in `api/types.go`? Perhaps need to be re-exported?

Reply:
this whole file will now disappear yes